app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
+          #add the owner back to the member_ids array
+          mi_array.unshift(@application.owner.owner_ids[0])
+          @application.owner.member_ids = mi_array if @application.owner.member_ids != mi_array
+        end
+        return true
+      else
+        @application.errors.add(:owner, 'Member Ids must be a space separated list of integers')
+        return false
+      end
+    end
+
+    def valid_member_ids(member_ids)
+      if member_ids.nil? || member_ids.empty?
+        return true
+      end
+      !!member_ids.match(NUMBERS_ONLY_REGEX)
+    end
+
+    def set_member_ids
+      #remove the owner from the array since it cannot be removed by users
+      ids = @application.owner.member_ids
+      owner_id = @application.owner.owner_ids[0]
+      ids.delete(owner_id)
```

Comment:
Are you intending to modify the state of `@application.owner.member_ids` here?  `ids` is a reference to that array so I think that's what is happening.  E.g. see this test on the Ruby console:

```ruby
irb(main):002:0> hash = {a: [1,2]}
=> {:a=>[1, 2]}
irb(main):003:0> a_ref = hash[:a]
=> [1, 2]
irb(main):004:0> a_ref.delete(2)
=> 2
irb(main):005:0> hash
=> {:a=>[1]}
```

Reply:
Also what is going on with the owner ID and member ID -- that is not clear to me.  Whatever it is, can you exercise that code in a spec?

Reply:
I changed the code to make a duplicate of the member_id array instead of changing it as you illustrated. I don't know how I can write a test for this. I'm open to suggestions.

Reply:
Can you say why the owner ID / member ID stuff is happening?  @Dantemss and I think something here is not right (e.g. he said "there's no reason the first owner has to be a member too as far as I remember")

Reply:
I was mainly doing that because of how the data looked on my local in the rails console. The owner was always first. I changed it so it is just added back.